prov/sockets/include/sock.h

```diff
@@ -869,6 +869,8 @@ struct sock_pe_entry {
 	struct dlist_entry ctx_entry;
 	struct ofi_ringbuf comm_buf;
 	size_t cache_sz;
+
+	ofi_atomic32_t completion_reported;
```

Comment:
I'm hesitant to add yet another state variable to the pe_entry.  We already have is_complete, is_error, and mr_checked.  There should be some way we can combine these into a single state machine.  is_complete is particularly odd, since I can't easily figure out what it means.  :/  It's set to 1 in several places after a completion is reported, but not all places.  We also need to ensure that completions aren't being reported multiple times when counters are used, and not just in the cq case.